CHANGES

```diff
@@ -29,6 +29,25 @@
      TODO(TLS1.3): Remove the above note before final release
      [Matt Caswell]
 
+  *) Grand redesign of the OpenSSL random generator
+
+     The default RAND method now utilizes an AES-CTR DRBG according to
+     NIST standard SP 800-90Ar1. The new random generator is essentially
+     a port of the default random generator from the OpenSSL FIPS 2.0
+     object module. It is a hybrid deterministic random bit generator
+     using an AES-CTR bit stream and which seeds and reseeds itself
+     automatically using trusted system entropy sources.
+
+     Some of its new features are:
+      o Support for multiple DRBG instances with seed chaining.
+      o Add a public DRBG instance for the default RAND method
+      o Add a dedicated DRBG instance for generating long term private keys.
+      o Make the DRBG instances fork-safe
+      o Keep all global DRBG instances on the secure heap if it is enabled.
+      o Add a DRBG instance to every SSL instance for lock free operation
+        and to increase unpredictability.
```

Comment:
I'm not comfortable with this line.  A DRBG is supposed to be unpredictable.  Does _separation_ work better than _unpredictability_ here?

This isn't a firm request to change.

Reply:
I was tempted to remove the second part of the sentence but then I left it as it is.  After all, a _deterministic_ random generator can never be absolutely unpredictable. The random bit stream can leak some information over a long time, e.g. by weakness of design or by security issues being exploited. Even the NIST standard has the notion of 'prediction resistance'. What I had in mind was a webserver with a lot of throughput, were the presence of many DRBG instances which reseed often drastically reduces the amount of random output that can be observed from a single DRBG instance.